component/stream/read_write.go

```diff
@@ -208,12 +210,23 @@ func (rw *ReadWriteCache) RenameFile(options internal.RenameFileOptions) error {
 	return nil
 }
 
+func (rw *ReadWriteCache) FlushFile(options internal.FlushFileOptions) error {
+	// log.Trace("Stream::FlushFile : name=%s, handle=%d", options.Handle.Path, options.Handle.ID)
+	err := rw.NextComponent().FlushFile(options)
+	if err != nil {
+		log.Err("Stream::FlushFile : error flushing file %s [%s]", options.Handle.Path, err.Error())
```

Comment:
If error was there in flush then we shall not clean up the dirty flag, and on close we shall reattempt flush.

Reply:
that sounds like strange behavior, is there a reason we follow such an approach? retry logic should be handled by storage, it seems odd to have a retry on close.

Reply:
in case of any timeout or any other error if flush fails this just gives us a second retry at flush during close. So in flush case if it fails we dont clear the flag and in close we always check if file is still dirty then first call flush and then do close handling. You can refer file-cache close code for this.

Reply:
done.